lib/chef/provisioning/lxc_driver/lxc_transport.rb

```diff
@@ -8,149 +8,153 @@
-              server = TCPServer.new(host, uri.port)
-              proxy = LXC::Extra::ProxyClientSide.new(channel, server)
-              proxy.start
-            rescue
-              Chef::Log.error("ERROR in proxy (container side): #{$!}\n#{$!.backtrace.join("\n")}")
-              raise
+              cmd.run_command
+              LXCExecuteResult.new(command, {}, cmd.stdout, cmd.stderr, cmd.exitstatus)
+            rescue Errno::ENOENT => e
+              LXCExecuteResult.new(command, {} , '', e.message, 2)
             end
           end
+        end
 
-          # Start the host side of the proxy, which contacts the real server
-          thread = Thread.new do
-            proxy = LXC::Extra::ProxyServerSide.new(channel) do
-              TCPSocket.new(host, uri.port)
+        def make_url_available_to_remote(local_url)
+          uri = URI(local_url)
+          host = Socket.getaddrinfo(uri.host, uri.scheme, nil, :STREAM)[0][3]
+          if host == '127.0.0.1' || host == '::1'
+            unless @port_forwards[uri.port]
+
+              Chef::Log.debug("Forwarding container port #{uri.port} to local port #{uri.port}")
+              # Create the channel that will let the container and the host talk to each other
+              channel = LXC::Extra::Channel.new
+
+              # Start the container side of the proxy, listening for client connections
+              pid = container.attach do
+                begin
+                  server = TCPServer.new(host, uri.port)
+                  proxy = LXC::Extra::ProxyClientSide.new(channel, server)
+                  proxy.start
+                rescue
+                  Chef::Log.error("ERROR in proxy (container side): #{$!}\n#{$!.backtrace.join("\n")}")
+                  raise
+                end
+              end
+
+              # Start the host side of the proxy, which contacts the real server
+              thread = Thread.new do
+                proxy = LXC::Extra::ProxyServerSide.new(channel) do
+                  TCPSocket.new(host, uri.port)
+                end
+                proxy.start
+              end
+
+              Chef::Log.debug("Forwarded #{uri.port} on container #{name} to local port #{uri.port}.  Container listener id PID #{pid}")
+
+              @port_forwards[uri.port] = [ pid, thread, channel ]
             end
-            proxy.start
-          end
-
-          Chef::Log.debug("Forwarded #{uri.port} on container #{name} to local port #{uri.port}.  Container listener id PID #{pid}")
 
-          @port_forwards[uri.port] = [ pid, thread, channel ]
+          end
+          local_url
         end
 
-      end
-      local_url
-    end
-
-    def read_file(path)
-      if File.exists?(container_path(path))
-        File.read(container_path(path))
-      end
-    end
-
-    def download_file(path, local_path)
-      Chef::Log.debug("Copying file #{path} from #{name} to local #{local_path}")
-      FileUtils.cp_r(container_path(path), local_path)
```

Comment:
I'm assuming the reason for this change is so that we can deal with containers whose filesystems aren't mounted on the host?  Is that even possible?